### ISSUE TITLE
Try to use the CSCS spack build cache in CI 

### ISSUE THREAD

**Issue by aurianer** [pika-org/pika#789]
Try to use the CSCS spack build cache in CI once #32 is done.

**Comment by msimberg** [comment]
I would keep this open as it's sufficiently different than what was implemented in #962 and may still be useful to do at some point. However, it's lower priority now that #962 is merged.

**Comment by msimberg** [comment]
> What about using https://cache.spack.io/tag/develop-2024-09-22/?stack=developer-tools-manylinux2014? I think it's pretty straightforward to add

Can you try reviving https://github.com/pika-org/pika/pull/1129 with a new cache? I don't even think you need to choose the stack, but it's ok to use the generic develop-yyyy-mm-dd cache. If I remember correctly the HIP build issue is now actually fixed, so we should be able to use the cache on all pipelines.

The above is still a read-only cache, while this issue is about a read-write cache. A read-only cache shouldn't hurt though so it's worth trying it again now assuming the HIP build issue is actually fixed.
